### ISSUE TITLE
Could not revert revision, due to database integrity errors.

### ISSUE THREAD

**Issue by oppianmatt** [etianen/django-reversion#349]
It would be nice to see the error so we could fix it.

I had to delve even deeper into the models and get the object directly to save and see what the issue was.

```
deleted_list.first().revision.version_set.all()[0].object_version.object
```

saw it was a foreign key (that could be set to null) trying to be restored to a value that didn't exist. I set it to null manually and saved the object. It seemed to work.

saw https://github.com/etianen/django-reversion/issues/86 but that wasn't the issue

**Comment by GaretJax** [comment]
I agree with @oppianmatt on this one, ideally, the raised RevertError would carry an attribute referencing the original exception for introspection.

P.S.: I got this while working on https://github.com/edoburu/django-parler/issues/21

**Comment by etianen** [comment]
Is there a “standard” way of referencing an exception that caused another in Python?

Certainly in Python 3, this is baked into the exception handling logic, and happens automatically.

**Comment by joshkel** [comment]
I ran into an internal server error similar to @skrzypek's today. In my case, the integrity error was caused because our CustomUser model had a uniqueness constraint on the email column, so if a user was deleted then recreated, then trying to view the deleted user using django-reversion's "Recover deleted users" operation would fail.

Can I use one of the existing signals to handle conflicts like this? Or is there a better approach?

> Is there a “standard” way of referencing an exception that caused another in Python? Certainly in Python 3, this is baked into the exception handling logic, and happens automatically.

I hadn't dealt with one exception causing another in Python before, but I did a bit of reading:

* If I understand [safe_revert](https://github.com/etianen/django-reversion/blob/feebd690c250344dc015a2c756d1884192506dcd/src/reversion/models.py#L35) correctly, it won't happen automatically in this case, because the code catches the IntegrityError, continues execution, and later throws the RevertError.
* If you want to explicitly chain exceptions (mark one exception as the cause of another), then the syntax would be `raise RevertError(msg) from e` (as documented in [PEP 3134](https://www.python.org/dev/peps/pep-3134/)).
* It looks like the [python-future library](http://python-future.org/compatible_idioms.html#raising-exceptions) can take care of explicit exception chaining in a Python 2-compatible fashion.

If you're willing to add python-future as a dependency, I can submit a PR.
